src/main/scala/mesosphere/marathon/core/appinfo/TaskCounts.scala

```diff
@@ -13,16 +13,34 @@ import org.apache.mesos.{ Protos => mesos }
   * @param tasksUnhealthy snapshot of the number of unhealthy tasks (does not include tasks without health info)
   */
 case class TaskCounts(
-  tasksStaged: Int,
-  tasksRunning: Int,
-  tasksHealthy: Int,
-  tasksUnhealthy: Int)
+    tasksStaged: Int,
+    tasksRunning: Int,
+    tasksHealthy: Int,
+    tasksUnhealthy: Int) {
+  def +(counts: TaskCounts): TaskCounts = {
+    copy(
+      tasksRunning = tasksRunning + counts.tasksRunning,
+      tasksStaged = tasksStaged + counts.tasksStaged,
+      tasksHealthy = tasksHealthy + counts.tasksHealthy,
+      tasksUnhealthy = tasksUnhealthy + counts.tasksUnhealthy
+    )
+  }
+
+  def -(counts: TaskCounts): TaskCounts = {
+    copy(
+      tasksRunning = tasksRunning - counts.tasksRunning,
+      tasksStaged = tasksStaged - counts.tasksStaged,
+      tasksHealthy = tasksHealthy - counts.tasksHealthy,
+      tasksUnhealthy = tasksUnhealthy - counts.tasksUnhealthy
```

Comment:
I see. I added the comment before I got to the usage; from that perspective it's ok. The only strange thing is that TaskCounts can actually be negative – yet I don't think an Exception would be any help here so just leave it like that.